test/testBody.cpp

```diff
@@ -0,0 +1,155 @@
+	EXPECT_EQ(request.body, "hello \nworld!\n");
+}
+
+// INVALID BODY TEST SUITE
+
+TEST(RequestParser_NonValidBody, DifferingChunkSize) {
+	RequestParser	p;
+    HTTPRequest request;
+
+	EXPECT_THROW(p.parseHttpRequest("GET /search?query=openai&year=2024#conclusion HTTP/1.1\r\nHost: www.example.com\r\nTransfer-Encoding: gzip, chunked\r\n\r\n1\r\nhello \r\n6\r\nworld!\r\n0\r\n\r\n", request), std::runtime_error);
+    p.clearRequest(request);
+    p.clearParser();
+    try{
+        p.parseHttpRequest("GET /search?query=openai&year=2024#conclusion HTTP/1.1\r\nHost: www.example.com\r\nTransfer-Encoding: gzip, chunked\r\n\r\n1\r\nhello \r\n6\r\nworld!\r\n0\r\n\r\n", request);
+    }
+    catch(const std::runtime_error& e) {
+        std::cerr << e.what() << std::endl;
+    }
+}
+
+TEST(RequestParser_NonValidBody, DifferingContentLength) {
+	RequestParser	p;
+    HTTPRequest request;
+
+	EXPECT_THROW(p.parseHttpRequest("GET /search?query=openai&year=2024#conclusion HTTP/1.1\r\nHost: www.example.com\r\nContent-Length: 3\r\n\r\nhello \r\nworld!\r\n", request), std::runtime_error);
+    p.clearRequest(request);
+    p.clearParser();
+    try{
+        p.parseHttpRequest("GET /search?query=openai&year=2024#conclusion HTTP/1.1\r\nHost: www.example.com\r\nContent-Length: 3\r\n\r\nhello \r\nworld!\r\n", request);
+    }
+    catch(const std::runtime_error& e) {
+        std::cerr << e.what() << std::endl;
+    }
+}
+
+TEST(RequestParser_NonValidBody, MissingCRLFInChunk) {
+	RequestParser	p;
+    HTTPRequest request;
+
+	EXPECT_THROW(p.parseHttpRequest("GET /search?query=openai&year=2024#conclusion HTTP/1.1\r\nHost: www.example.com\r\nTransfer-Encoding: gzip, chunked\r\n\r\n6\r\nhello 6\r\nworld!\r\n0\r\n\r\n", request), std::runtime_error);
+    p.clearRequest(request);
+    p.clearParser();
+    try{
+        p.parseHttpRequest("GET /search?query=openai&year=2024#conclusion HTTP/1.1\r\nHost: www.example.com\r\nTransfer-Encoding: gzip, chunked\r\n\r\n6\r\nhello 6\r\nworld!\r\n0\r\n\r\n", request);
+    }
+    catch(const std::runtime_error& e) {
+        std::cerr << e.what() << std::endl;
+    }
+}
+
+TEST(RequestParser_NonValidBody, MissingCRInChunk) {
+	RequestParser	p;
+    HTTPRequest request;
+
+	EXPECT_THROW(p.parseHttpRequest("GET /search?query=openai&year=2024#conclusion HTTP/1.1\r\nHost: www.example.com\r\nTransfer-Encoding: gzip, chunked\r\n\r\n6\r\nhello \n6\r\nworld!\r\n0\r\n\r\n", request), std::runtime_error);
+    p.clearRequest(request);
+    p.clearParser();
+    try{
+        p.parseHttpRequest("GET /search?query=openai&year=2024#conclusion HTTP/1.1\r\nHost: www.example.com\r\nTransfer-Encoding: gzip, chunked\r\n\r\n6\r\nhello \n6\r\nworld!\r\n0\r\n\r\n", request);
+    }
+    catch(const std::runtime_error& e) {
+        std::cerr << e.what() << std::endl;
+    }
+}
+
+// PREVIOUS TEST CODE
```

Comment:
Do we still need this? (repeated for all three test files)

Reply:
I'd like to keep it as reference, at least for now.